log/log.go

```diff
@@ -1,72 +1,108 @@
+	}
+
+	logLevel := resolveLogLevelFromEnv()
+	zerolog.SetGlobalLevel(logLevel)
+}
 
 func Info(args ...interface{}) {
-	glog.InfoDepth(1, args...)
+	log.Info().Msgf("%s", args...)
 }
 
 func Infof(format string, args ...interface{}) {
-	glog.InfoDepth(1, fmt.Sprintf(format, args...))
+	log.Info().Msgf(format, args...)
 }
 
 func Warning(args ...interface{}) {
-	glog.WarningDepth(1, args...)
+	log.Warn().Msgf("%s", args...)
 }
 
 func Warningf(format string, args ...interface{}) {
-	glog.WarningDepth(1, fmt.Sprintf(format, args...))
+	log.Warn().Msgf(format, args...)
 }
 
 func Error(args ...interface{}) {
-	glog.ErrorDepth(1, args...)
+	log.Error().Msgf("%s", args...)
 }
 
 func Errorf(format string, args ...interface{}) {
-	glog.ErrorDepth(1, fmt.Sprintf(format, args...))
+	log.Error().Msgf(format, args...)
 }
 
 // Debug will log a message at verbose level 4 and will ensure the caller's stack frame is used
 func Debug(args ...interface{}) {
-	if glog.V(debug) {
-		glog.InfoDepth(1, "DEBUG: "+fmt.Sprint(args...)) // 1 == depth in the stack of the caller
-	}
+	log.Debug().Msgf("%s", args...)
 }
 
 // Debugf will log a message at verbose level 4 and will ensure the caller's stack frame is used
 func Debugf(format string, args ...interface{}) {
-	if glog.V(debug) {
-		glog.InfoDepth(1, fmt.Sprintf("DEBUG: "+format, args...)) // 1 == depth in the stack of the caller
-	}
+	log.Debug().Msgf(format, args...)
 }
 
 func IsDebug() bool {
-	return bool(glog.V(debug))
+	return zerolog.GlobalLevel() == zerolog.DebugLevel
 }
 
 // Trace will log a message at verbose level 5 and will ensure the caller's stack frame is used
 func Trace(args ...interface{}) {
-	if glog.V(trace) {
-		glog.InfoDepth(1, "TRACE: "+fmt.Sprint(args...)) // 1 == depth in the stack of the caller
-	}
+	log.Trace().Msgf("%s", args...)
 }
 
 // Tracef will log a message at verbose level 5 and will ensure the caller's stack frame is used
 func Tracef(format string, args ...interface{}) {
-	if glog.V(trace) {
-		glog.InfoDepth(1, fmt.Sprintf("TRACE: "+format, args...)) // 1 == depth in the stack of the caller
-	}
+	log.Trace().Msgf(format, args...)
 }
 
 func IsTrace() bool {
-	return bool(glog.V(trace))
+	return zerolog.GlobalLevel() == zerolog.TraceLevel
+}
+
+func Fatal(args ...interface{}) {
+	log.Fatal().Msgf("%s", args...)
+}
+
+func Fatalf(format string, args ...interface{}) {
+	log.Fatal().Msgf(format, args...)
+}
+
+// Resolves the environment settings for the log level. Considers the verbose_mode from server version <=1.25.
+func resolveLogLevelFromEnv() zerolog.Level {
+	logLevel, isDefined := os.LookupEnv("LOG_LEVEL")
+
+	if !isDefined {
+		return zerolog.InfoLevel
+	}
+
+	switch logLevel {
+	case "0":
+		return zerolog.InfoLevel
+	case "1":
+		return zerolog.WarnLevel
+	case "2":
+		return zerolog.ErrorLevel
+	case "3":
+		return zerolog.FatalLevel
+	case "4":
+		return zerolog.DebugLevel
+	case "5":
+		return zerolog.TraceLevel
```

Comment:
I think these levels are incorrect. Based on the current [Kiali CR documentation](https://github.com/kiali/kiali-operator/blob/v1.24/deploy/kiali/kiali_cr.yaml#L355-L358), where it says,

```
Determines which priority levels of log messages Kiali will output.
# Typical values are "3" for INFO and higher priority, "4" for DEBUG and higher priority.
```
This is based on the glog "V" levels we used before, higher numbers get noisier as you go up. So in Kiali, 3 is INFO, 4 is DEBUG and 5 is TRACE , with 2 being WARN, and 1 being ERROR and 0 being FATAL.
So Kiali users will be using lower numbers to make the log quieter.

```suggestion
	case "0":
		return zerolog.FatalLevel
	case "1":
		return zerolog.ErrorLevel
	case "2":
		return zerolog.WarnLevel
	case "3":
		return zerolog.InfoLevel
	case "4":
		return zerolog.DebugLevel
	case "5":
		return zerolog.TraceLevel
```